### PR TITLE
Avoid echoing onto a captured FD

### LINKED ISSUE #859: IPKernelApp configs capture_fd_output and quiet conflict and break ipykernel
When `c.IPKernelApp.capture_fd_output` is True (default True) and `c.IPKernelApp.quiet` if False (default True), printing anything causes an infinite output in the cell. `quiet = True` causes outputs to get directed to fd, which then gets picked up by the `capture_fd_output` logic and re-printed. This loop continues until the process is killed.

### PR DESCRIPTION
**Pull request by minrk**
When echoing output onto stderr, it should be onto the copy, not the redirected FD, so it only goes to the terminal instead of making an infinite loop, echoing and then capturing and then re-echoing the same output.

I've verified that this works, I just need to work out the tests without messing up pytest output itself.

closes #859

### CODE REVIEW

**Comment by garlandz-db** [comment]
@minrk thanks for fixing this! lmk how i can help drive this to completion

**Comment by minrk** [comment]
Finally got all actual tests to pass!

File: ipykernel/tests/test_io.py [hunk 1]
```diff
@@ -10,20 +15,29 @@
...
-    """Test that wrapped stdout has the same API as a normal TextIO object"""
-    session = Session()
+@pytest.fixture
+def ctx():
     ctx = zmq.Context()
-    pub = ctx.socket(zmq.PUB)
-    thread = IOPubThread(pub)
-    thread.start()
+    return ctx
+    # yield ctx
```
[the remark this answers is not in the document]
Comment (minrk) [reply ipykernel/tests/test_io.py]: Yes, should be yield. Not sure what I was doing, there. Thanks!

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**